railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -46,6 +46,8 @@ def build_stack
           middleware.use ::ActionDispatch::RequestId
           middleware.use ::ActionDispatch::RemoteIp, config.action_dispatch.ip_spoofing_check, config.action_dispatch.trusted_proxies
 
+          middleware.use ::ActionDispatch::ContentSecurityPolicy::Middleware, config.warn_on_no_content_security_policy
```

Comment:
You are changing the position of this middleware to be earlier in the stack before the Rack::Logger. I don't think we should do that. What we want is to move before the public exceptions middleware so this should be probably after the Rack::Logger.